specs/_features/epbs/beacon-chain.md

```diff
@@ -0,0 +1,779 @@
+
+```python
+def is_active_validator_at_index(state: BeaconState, index: ValidatorIndex) -> Bool:
+    if index < len(state.validators):
+        validator = state.validators[index]
+    else:
+        validator = state
+
+```
+
+*Note*: The function `get_eligible_validator_indices` is modified to account for builders.
+
+```python
+def get_eligible_validator_indices(state: BeaconState) -> Sequence[ValidatorIndex]:
+    previous_epoch = get_previous_epoch(state)
+    return [
+        ValidatorIndex(index) for index, v in enumerate(state.validators + state.builders)
+        if is_active_validator(v, previous_epoch) or (v.slashed and previous_epoch + 1 < v.withdrawable_epoch)
+    ]
+```
+
+
+### Execution engine
+
+#### Request data
+
+##### New `NewInclusionListRequest`
+
+```python
+@dataclass
+class NewInclusionListRequest(object):
+    inclusion_list: List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+  
+#### Engine APIs
+
+#### Modified `notify_new_payload`
+*Note*: the function notify new payload is modified to raise an exception if the payload is not valid, and to return the list of transactions that remain valid in the inclusion list
+
+```python
+def notify_new_payload(self: ExecutionEngine, execution_payload: ExecutionPayload) -> List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]:
+    """
+    Raise an exception if ``execution_payload`` is not valid with respect to ``self.execution_state``. 
+    Returns the list of transactions in the inclusion list that remain valid after executing the payload. That is
+    it is guaranteed that the transactions returned in the list can be executed in the exact order starting from the 
+    current ``self.execution_state``.
+    """
+    ...
+```
+
+#### Modified `verify_and_notify_new_payload`
+*Note*: the function `verify_and_notify_new_payload` is modified so that it returns the list of transactions that remain valid in the forward inclusion list. It raises an exception if the payload is not valid. 
+
+```python
+def verify_and_notify_new_payload(self: ExecutionEngine,
+                                  new_payload_request: NewPayloadRequest) -> List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]:
+    """
+    Raise an exception if ``execution_payload`` is not valid with respect to ``self.execution_state``. 
+    Returns the list of transactions in the inclusion list that remain valid after executing the payload. That is
+    it is guaranteed that the transactions returned in the list can be executed in the exact order starting from the 
+    current ``self.execution_state``. 
+    """
+    assert self.is_valid_block_hash(new_payload_request.execution_payload)
+    return self.notify_new_payload(new_payload_request.execution_payload)
+```
+
+#### New `notify_new_inclusion_list`
+
+```python
+def notify_new_inclusion_list(self: ExecutionEngine,
+                              inclusion_list_request: NewInclusionListRequest) -> bool:
```

Comment:
I'm inclined to say the current argument is good enough, and we don't need `parentBlockHash` as an argument. The edge case I'm thinking of is a gas limit that can change in-flight and the EL needs to know which IL it's built on top of